service/src/main/java/bio/terra/workspace/service/danglingresource/DanglingResourceCleanupService.java

```diff
@@ -0,0 +1,235 @@
+  private final CronjobDao cronjobDao;
+  private final SamService samService;
+  private final CrlService crlService;
+  private final ScheduledExecutorService scheduler;
+
+  private final List<WsmResourceType> DANGLING_RESOURCE_TYPES =
+      List.of(
+          WsmResourceType.CONTROLLED_GCP_AI_NOTEBOOK_INSTANCE,
+          WsmResourceType.CONTROLLED_GCP_GCE_INSTANCE,
+          WsmResourceType.CONTROLLED_GCP_DATAPROC_CLUSTER);
+
+  @Autowired
+  public DanglingResourceCleanupService(
+      DanglingResourceCleanupConfiguration configuration,
+      ControlledResourceService controlledResourceService,
+      JobService jobService,
+      ResourceDao resourceDao,
+      CronjobDao cronjobDao,
+      SamService samService,
+      CrlService crlService) {
+    this.configuration = configuration;
+    this.controlledResourceService = controlledResourceService;
+    this.jobService = jobService;
+    this.resourceDao = resourceDao;
+    this.cronjobDao = cronjobDao;
+    this.samService = samService;
+    this.crlService = crlService;
+    this.scheduler = Executors.newScheduledThreadPool(1);
+  }
+
+  @PostConstruct
+  public void startStatusChecking() {
+    if (configuration.isEnabled()) {
+      // Per scheduleAtFixedRate documentation, if a single execution runs longer than the polling
+      // interval, subsequent executions may start late but will not concurrently execute.
+      scheduler.scheduleAtFixedRate(
+          this::cleanupResourcesSuppressExceptions,
+          configuration.getStartupWait().toSeconds(),
+          configuration.getPollingInterval().toSeconds(),
+          TimeUnit.SECONDS);
+    }
+  }
+
+  /**
+   * Run {@code cleanupResources}, suppressing all thrown exceptions. This is helpful as {@code
+   * ScheduledExecutorService.scheduleAtFixedRate} will stop running if any execution throws an
+   * exception. Suppressing these exceptions ensures we do not stop cleaning up resources if a
+   * single run fails.
+   */
+  public void cleanupResourcesSuppressExceptions() {
+    try {
+      cleanupResources();
+    } catch (Exception e) {
+      LoggingUtils.logAlert(
+          logger,
+          "Unexpected error during danglingResourceCleanup execution, see stacktrace below");
+      logger.error("danglingResourceCleanup stacktrace: ", e);
+    }
+  }
+
+  private void cleanupResources() {
+    if (!configuration.isEnabled()) {
+      return;
+    }
+    logger.info("Beginning dangling resource cleanup cronjob");
+    // Use a one-second shorter duration here to ensure we don't skip a run by moving slightly too
+    // quickly.
+    Duration claimTime = configuration.getPollingInterval().minus(Duration.ofSeconds(1));
+    // Attempt to claim the latest run of this job to ensure only one pod runs the cleanup job.
+    if (!cronjobDao.claimJob(DANGLING_RESOURCE_CLEANUP_JOB_NAME, claimTime)) {
+      logger.info("Another pod has executed this job more recently. Ending resource cleanup.");
```

Comment:
nit here and in other cronjobs: since we're running multiple cronjobs, can you include the job name in this log message? (e.g. something like "Another pod has executed dangling resource cleanup more recently, skipping this execution")